src/appmixer/servicenow/record/UpdateRecord/component.json

```
@@ -1,61 +1,122 @@
+                    "tableName": {
                         "type": "text",
-                        "label": "Sys ID",
+                        "label": "Table name",
                         "index": 1,
-                        "tooltip": "Sys ID of the record to update.",
-                        "group": "options"
+                        "tooltip": "Name of the table from which to retrieve the record."
+                    },
+                    "sys_id": {
+                        "type": "text",
+                        "label": "Sys ID",
+                        "index": 2,
+                        "tooltip": "Sys ID of the record to update."
+                    },
+                    "record": {
+                        "type": "textarea",
+                        "label": "Record",
+                        "index": 3,
+                        "tooltip": "JSON object containing field values for the new record. Example: <code>{\"email\":\"foo@bar.baz\",\"name\":\"Foo\"}</code>. This takes precedence over the <code>pairs</code> field(s)."
+                    },
+                    "pairs": {
+                        "type": "expression",
+                        "label": "Fields",
+                        "tooltip": "Fields to update a record.",
+                        "exclusiveFields": [
+                            "field"
+                        ],
+                        "index": 4,
+                        "levels": [
+                            "AND"
+                        ],
+                        "fields": {
+                            "field": {
+                                "type": "text",
+                                "label": "Field",
+                                "index": 1,
+                                "tooltip": "The field/column name."
+                            },
+                            "value": {
+                                "type": "text",
+                                "label": "Value",
+                                "index": 2,
+                                "tooltip": "The value for the field."
+                            }
+                        }
+                    },
+                    "sysparm_display_value": {
+                        "type": "toggle",
+                        "label": "Display value",
+                        "index": 5,
+                        "tooltip": "Return field display values (true), actual values (false), or both (all) (default: false)"
+                    },
+                    "sysparm_exclude_reference_link": {
+                        "type": "toggle",
+                        "label": "Exclude reference link",
+                        "index": 6,
+                        "tooltip": "True to exclude Table API links for reference fields (default: false)"
                     },
                     "sysparm_fields": {
                         "type": "text",
                         "label": "Fields",
```

```suggestion
                    "sysparm_fields": {
                        "type": "text",
                        "label": "Return Fields",
```
just an idea - renaming it to "Return Fields" to distinguish it from update fields above